crates/protocol/protocol/src/deposits.rs

```diff
@@ -128,40 +139,56 @@ pub fn decode_deposit(block_hash: B256, index: usize, log: &Log) -> Result<Bytes
     // | 32     | [0; 24] . {U64 big endian, hex encoded length}  |
     // ------------------------------------------------------------
 
-    let opaque_content_offset: U64 = U64::try_from_be_slice(&log.data.data[24..32]).ok_or(
-        DepositError::InvalidOpaqueDataOffset(Bytes::copy_from_slice(&log.data.data[24..32])),
+    let opaque_content_offset: U256 = U256::try_from_be_slice(&log.data.data[0..32]).ok_or(
+        DepositError::InvalidOpaqueDataOffset(Bytes::copy_from_slice(&log.data.data[0..32])),
     )?;
-    if opaque_content_offset != U64::from(32) {
+    if opaque_content_offset != U256::from(32) {
         return Err(DepositError::InvalidOpaqueDataOffset(Bytes::copy_from_slice(
-            &log.data.data[24..32],
+            &log.data.data[0..32],
         )));
     }
 
     // The next 32 bytes indicate the length of the opaqueData content.
-    let opaque_content_len =
-        u64::from_be_bytes(log.data.data[56..64].try_into().map_err(|_| {
-            DepositError::InvalidOpaqueDataLength(Bytes::copy_from_slice(&log.data.data[56..64]))
-        })?);
-    if opaque_content_len as usize > log.data.data.len() - 64 {
-        return Err(DepositError::OpaqueDataOverflow(
-            opaque_content_len as usize,
-            log.data.data.len() - 64,
-        ));
+    let opaque_content_len: U256 = U256::try_from_be_slice(&log.data.data[32..64]).ok_or(
```

Comment:
Do we ever expect converting 32 bytes into a U256 to fail?

```suggestion
    let opaque_content_len: U256 = U256::from_be_slice(&log.data.data[32..64]).ok_or(
```